test/agents/test_conversational_agent.py

```diff
@@ -1,56 +1,90 @@
 import pytest
-from unittest.mock import MagicMock, patch
+from unittest.mock import MagicMock
 
+from haystack.errors import AgentError
+from haystack.agents.base import Tool
 from haystack.agents.conversational import ConversationalAgent
 from haystack.agents.memory import ConversationSummaryMemory, ConversationMemory, NoMemory
 from haystack.nodes import PromptNode
 
 
+@pytest.fixture
+def prompt_node():
+    prompt_node = PromptNode()
+    return prompt_node
```

Comment:
We can't use this fixture in unit tests as it will load a model, why not using [`MockPromptNode`](https://github.com/deepset-ai/haystack/blob/d8a4f203793332b48f14fa8286397f25cddcc20e/test/conftest.py#L394)?

Reply:
MockPromptNode has a different logic than PromptNode in fetching the correct prompt. Therefore, MockPromptNode fails to fetch `conversation-summary` and ConversationalAgent related prompts. 
I can open a new issue to update MockPromptNode rather than changing it with this PR

Reply:
Then we need to mock the model loading here. Otherwise, once #5088 is merged, all unit tests using this `prompt_node` fixture will fail due to making an http request.

Reply:
@patch("haystack.nodes.prompt.prompt_node.PromptModel") can be used here. See how other tests use it in test_prompt_node.py